client/homescreen/activity-panel/orders/index.js

```diff
@@ -291,93 +341,4 @@ OrdersPanel.defaultProps = {
 	isRequesting: false,
 };
 
-OrdersPanel.contextType = CurrencyContext;
-
-export default withSelect( ( select, props ) => {
-	const { countUnreadOrders, orderStatuses } = props;
-	const { getItems, getItemsError } = select( ITEMS_STORE_NAME );
-	const { getReportItems, getReportItemsError, isResolving } = select(
-		REPORTS_STORE_NAME
-	);
-
-	if ( ! orderStatuses.length && countUnreadOrders === 0 ) {
-		return { isRequesting: false };
-	}
-
-	// Query the core Orders endpoint for the most up-to-date statuses.
-	const actionableOrdersQuery = {
-		page: 1,
-		per_page: 5,
-		status: orderStatuses,
-		_fields: [ 'id', 'date_created_gmt', 'status' ],
-	};
-	/* eslint-disable @wordpress/no-unused-vars-before-return */
-	const actionableOrders = Array.from(
-		getItems( 'orders', actionableOrdersQuery ).values()
-	);
-	const isRequestingActionable = isResolving( 'getItems', [
-		'orders',
-		actionableOrdersQuery,
-	] );
-
-	if ( isRequestingActionable ) {
-		return {
-			isError: Boolean(
-				getItemsError( 'orders', actionableOrdersQuery )
-			),
-			isRequesting: isRequestingActionable,
-			orderStatuses,
-		};
-	}
-
-	// Retrieve the Order stats data from our reporting table.
-	const ordersQuery = {
-		page: 1,
-		per_page: 5,
-		extended_info: true,
-		match: 'any',
-		order_includes: map( actionableOrders, 'id' ),
-		status_is: orderStatuses,
-		_fields: [
-			'order_id',
-			'order_number',
-			'status',
-			'total_sales',
-			'total_formatted',
-			'extended_info.customer',
-			'extended_info.products',
-		],
-	};
-
-	const reportOrders = getReportItems( 'orders', ordersQuery ).data;
-	const isError = Boolean( getReportItemsError( 'orders', ordersQuery ) );
-	const isRequesting = isResolving( 'getReportItems', [
-		'orders',
-		ordersQuery,
-	] );
-	/* eslint-enable @wordpress/no-unused-vars-before-return */
-	let orders = [];
-
-	if (
-		countUnreadOrders === null ||
-		typeof reportOrders === 'undefined' ||
-		( reportOrders.length && ! actionableOrders.length ) ||
-		isRequesting
-	) {
-		return { isRequesting: true };
-	}
```

Comment:
This part has all been removed in the `useSelect` above to only use the `getItems` query, making use of the `actionableOrders` query that was already happening.